lldp_dcbx_nl.c

```diff
@@ -279,6 +279,7 @@ static int get_state(char *ifname, __u8 *state)
 		NLMSG_ALIGN(sizeof(struct dcbmsg)));
 
 	if (d->cmd != DCB_CMD_GSTATE) {
+		free(nlh);
 		return -EIO;
 	}
```

Comment:
This is good, but it might be nicer, for code consistency, to make it follow the pattern used in other functions in this file:
```suggestion
	if (d->cmd != DCB_CMD_GSTATE) {
		rval = -EIO;
        goto err_out;
	}
```
And then add the goto label just before the `free(nlh);`